Unify Formatting (#82)
C2Devel/terraform-provider-croccloud: website/docs/r/instance.md

The following arguments are supported:

* `affinity` - (Optional) The affinity setting for an instance on a dedicated host.
    * _Valid values_: `default`, `host`

>     * _Constraints:_ The parameter could be set to `host` only if `tenancy` is `host`.

Reviewer: Точка не нужна

Перепроверь все констрейнты тут, пожалуйста

* `ami` - (Optional) An image to use for the instance.
  If an image is specified in the launch template, the `ami` setting will override it.

>     * _Constraints:_ Required unless `launch_template` is specified.

Reviewer: Точка не нужна

* `associate_public_ip_address` - (Optional) Indicates whether to associate a public IP address with an instance in a VPC.
    * _Constraints:_ Conflicts with `network_interface`.
      The address will be assigned to the `eth0` interface only if there are free allocated external addresses.
      This operation is available only for instances running in the VPC and for new network interfaces
* `availability_zone` - (Optional) An availability zone to start the instance in.
* `disable_api_termination` - (Optional) If `true`, disables the possibility to terminate an instance via API.
* `ebs_block_device` - (Optional) One or more configuration blocks with additional EBS block devices to attach to the instance. The structure of this block and details on drift detection are [described below](#ebs-block-device). When accessing this as an attribute reference, it is a set of objects.

>     * _Constraints:_ Block device configurations are applied only when the resource is created.

Reviewer: Точка не нужна

    * _Constraints:_ Conflicts with `user_data`
* `user_data_replace_on_change` - (Optional) When used in combination with `user_data` or `user_data_base64` will trigger a destroy and recreate when set to `true`.
    * _Default value_: `false`
* `volume_tags` - (Optional) A map of tags to assign to root and EBS volumes when the instance is created.

~> **Note** Do not use `volume_tags` if you plan to manage block device tags outside the `aws_instance` configuration, such as using `tags` in an [`aws_ebs_volume`](ebs_volume.md) resource attached via [`aws_volume_attachment`](volume_attachment.md). Doing so will result in resource cycling and inconsistent behavior.

* `vpc_security_group_ids` - (Optional) A list of security group IDs to associate with.
    * _Constraints:_ Conflicts with `network_interface`

> ### EBS Block Device

Reviewer: ```suggestion
### ebs_block_device
```

Для остальных блоков тоже нужно поменять

The `ebs_block_device` block has the following structure:

* `delete_on_termination` - (Optional) Whether the volume should be destroyed on instance termination.
    * _Default value:_ `true`
* `device_name` - (Required) Name of the device to mount.
* `iops` - (Optional) Amount of provisioned IOPS.

>     * _Constraints:_ Only valid for the volume_type `io2`

Reviewer: ```suggestion
    * _Constraints:_ Only valid for the volume type `io2`
```

* `snapshot_id` - (Optional) ID of the snapshot to mount.

> * `tags` - (Optional) A map of tags to assign to the device.

Reviewer: ```suggestion
* `snapshot_id` - (Optional) The ID of the snapshot to mount.
* `tags` - (Optional) Map of tags to assign to the volume.
```

The `network_interface` block has the following structure:

* `delete_on_termination` - (Optional) Whether to delete the network interface on instance termination.
    *_Default value:_ `false`
    * _Constraints:_ Currently, the only valid value is `false`, as this option is only supported when creating new network interfaces during instance launching.
* `device_index` - (Required) Integer index of the network interface attachment.
* `network_interface_id` - (Required) ID of the network interface to attach.

### Launch Template

~> **Note** Launch template parameters will be used only once when the instance is created.
If you want to update existing instance you need to change parameters directly.
Updating the launch template specification will force a new instance.

Any other instance parameters that you specify will override the same parameters in the launch template.

The `launch_template` block has the following structure:

* `id` - The ID of the launch template.
* `name` - The name of the launch template.
* `version` - Template version.
    * _Valid values_:  A version number, `$Latest`, `$Default`
    * _Default value:_ `$Default`

> ### Root Block Device

Reviewer: Те же замечания, что и к `ebs_block_device`

The following arguments are supported:

* `affinity` - (Optional) The affinity setting for an instance on a dedicated host.
    * _Valid values_: `default`, `host`
    * _Constraints:_ The parameter could be set to `host` only if `tenancy` is `host`
* `ami` - (Optional) An image to use for the instance.
  If an image is specified in the launch template, the `ami` setting will override it.
    * _Constraints:_ Required unless `launch_template` is specified
* `associate_public_ip_address` - (Optional) Indicates whether to associate a public IP address with an instance in a VPC.
    * _Constraints:_ Conflicts with `network_interface`
      The address will be assigned to the `eth0` interface only if there are free allocated external addresses.
      This operation is available only for instances running in the VPC and for new network interfaces
* `availability_zone` - (Optional) An availability zone to start the instance in.
* `disable_api_termination` - (Optional) If `true`, disables the possibility to terminate an instance via API.

> * `ebs_block_device` - (Optional) One or more configuration blocks with additional EBS block devices to attach to the instance. The structure of this block and details on drift detection are [described below](#ebs-block-device). When accessing this as an attribute reference, it is a set of objects.

Reviewer: ```suggestion
* `ebs_block_device` - (Optional) One or more configuration blocks with additional EBS block devices to attach to the instance. The structure of this block and details on drift detection are [described below](#ebs_block_device). When accessing this as an attribute reference, it is a set of objects.
```

Нужно обновить и другие ссылки в этом файле

Author: обновил

* `ephemeral_block_device` - (Optional) One or more configuration blocks to customize ephemeral volumes on the instance. The structure of this block is [described below](#ephemeral_block_device). When accessing this as an attribute reference, it is a set of objects.
* `host_id` - (Optional) The ID of the dedicated host that the instance will be assigned to.
* `instance_initiated_shutdown_behavior` - (Optional) Shutdown behavior for the instance.
    * _Valid values_: `stop`, `terminate`
* `instance_type` - (Optional) The instance type to use for the instance. Updates to this field will trigger a stop/start of the EC2 instance.
* `key_name` - (Optional) Key name of the key pair to use for the instance; which can be managed using [the `aws_key_pair` resource](key_pair.md).
* `launch_template` - (Optional) Specifies a launch template to configure the instance. Parameters configured on this resource will override the corresponding parameters in the launch template.
  The structure of this block is [described below](#launch_template).
* `monitoring` - (Optional) If `true`, the launched EC2 instance will have detailed monitoring enabled.

> * `network_interface` - (Optional) Customize network interfaces to be attached at instance boot time. The structure of this block is [described below](#network_interfaces).

Reviewer: ```suggestion
* `network_interface` - (Optional) Customize network interfaces to be attached at instance boot time. The structure of this block is [described below](#network_interface).
```

    * _Constraints:_ Conflicts with `associate_public_ip_address`, `private_ip`, `secondary_private_ips`, `subnet_id`, `vpc_security_group_ids`
* `placement_group` - (Optional) Placement group to start the instance in.
* `private_ip` - (Optional) Private IP address to associate with the instance in a VPC.
    * _Constraints:_ Conflicts with `network_interface`
* `root_block_device` - (Optional) Root block device of the instance. The structure of this block is [described below](#root-block-device). When accessing this as an attribute reference, it is a list containing one object.

> * `secondary_private_ips` - (Optional) A list of secondary private IPv4 addresses to assign to the instance's primary network interface in a VPC.

Reviewer: Без артикля

    * _Constraints:_ Conflicts with `network_interface`
       Currently, only the primary private IP address can be specified.
* `source_dest_check` - (Optional) Controls if traffic is routed to the instance when the destination address does not match the instance.

>     * _Default value:_ `true`

Reviewer: Двоеточие попало в форматирование. В других местах тоже нужно проверить

Author: поправил, но надо ли

* `user_data` - (Optional) User data to provide when launching the instance. Do not pass gzip-compressed data via this argument; see `user_data_base64` instead. Updates to this field will trigger a stop/start of the EC2 instance by default. If the `user_data_replace_on_change` is set then updates to this field will trigger a destroy and recreate.
    * _Constraints:_ Conflicts with `user_data_base64`
* `user_data_base64` - (Optional) Can be used instead of `user_data` to pass base64-encoded binary data directly. Use this instead of `user_data` whenever the value is not a valid UTF-8 string. For example, gzip-encoded user data must be base64-encoded and passed via this argument to avoid corruption. Updates to this field will trigger a stop/start of the EC2 instance by default. If the `user_data_replace_on_change` is set then updates to this field will trigger a destroy and recreate.
    * _Constraints:_ Conflicts with `user_data`
* `user_data_replace_on_change` - (Optional) When used in combination with `user_data` or `user_data_base64` will trigger a destroy and recreate when set to `true`.
    * _Default value_: `false`

> * `volume_tags` - (Optional) A map of tags to assign to root and EBS volumes when the instance is created.

Reviewer: Без артикля

~> **Note** Do not use `volume_tags` if you plan to manage block device tags outside the `aws_instance` configuration, such as using `tags` in an [`aws_ebs_volume`](ebs_volume.md) resource attached via [`aws_volume_attachment`](volume_attachment.md). Doing so will result in resource cycling and inconsistent behavior.

> * `vpc_security_group_ids` - (Optional) A list of security group IDs to associate with.

Reviewer: Без артикля

    * _Constraints:_ Conflicts with `network_interface`

### ebs_block_device

The `ebs_block_device` block has the following structure:

* `delete_on_termination` - (Optional) Whether the volume should be destroyed on instance termination.
    * _Default value:_ `true`

> * `device_name` - (Required) Name of the device to mount.

Reviewer: Обязательный параметр должен быть выше

* `virtual_name` - (Optional) A name for the ephemeral device. Must match with the device name.

### network_interface

Each of the `network_interface` blocks attaches a network interface to an EC2 instance during boot time.
However, because the network interface is attached at boot-time, replacing/modifying the network interface **WILL** trigger a recreation of the EC2 instance.
If you should need at any point to detach/modify/re-attach a network interface to the instance, use the [`aws_network_interface`](network_interface.md) or [`aws_network_interface_attachment`](network_interface_attachment.md) resources instead.

The `network_interface` configuration block _does_, however, allow users to supply their own network interface to be used as the default network interface on an EC2 instance, attached at `eth0`.

The `network_interface` block has the following structure:

* `delete_on_termination` - (Optional) Whether to delete the network interface on instance termination.
    *_Default value:_ `false`
    * _Constraints:_ Currently, the only valid value is `false`, as this option is only supported when creating new network interfaces during instance launching

> * `device_index` - (Required) Integer index of the network interface attachment.

Reviewer: Обязательные параметры должны быть выше

> * `network_interface_id` - (Required) ID of the network interface to attach.

Reviewer: The ID of

### launch_template

~> **Note** Launch template parameters will be used only once when the instance is created.
If you want to update existing instance you need to change parameters directly.
Updating the launch template specification will force a new instance.

Any other instance parameters that you specify will override the same parameters in the launch template.

The `launch_template` block has the following structure:

* `id` - The ID of the launch template.
* `name` - The name of the launch template.
* `version` - Template version.

>     * _Valid values_:  A version number, `$Latest`, `$Default`

Reviewer: Двойной пробел

    * _Default value:_ `$Default`

### root_block_device

The `root_block_device` block has the following structure:

* `delete_on_termination` - (Optional) Whether the volume should be destroyed on instance termination.
    * _Default value:_ `true`
* `iops` - (Optional) Amount of provisioned IOPS.
    * _Constraints:_ Only valid for volume_type of `io2`

> * `tags` - (Optional) A map of tags to assign to the device.

Reviewer: Без артикля

* `volume_size` - (Optional) Size of the volume in GiB.
* `volume_type` - (Optional) Type of volume.
    * _Valid values_: `st2`, `gp2`, `io2`

## Attribute Reference

### Supported attributes

In addition to all arguments above, the following attributes are exported:

* `arn` - The Amazon Resource Name (ARN) of the instance.
* `instance_state` - The state of the instance.
    *_Valid values_:: `pending`, `running`, `shutting-down`, `terminated`, `stopping`, `stopped`
* `primary_network_interface_id` - The ID of the instance's primary network interface.
* `private_dns` - The private DNS name assigned to the instance. For EC2-VPC, this is only available if you've enabled DNS hostnames for your VPC.
* `public_dns` - The public DNS name assigned to the instance. For EC2-VPC, this is only available if you've enabled DNS hostnames for your VPC.
* `public_ip` - The public IP address assigned to the instance, if applicable.

  ~> **NOTE** If you are using [`aws_eip`](eip.md) with your instance, you should refer to the EIP's address directly and not use `public_ip` as this field will change after the EIP is attached.

> * `security_groups` - The list of security group names associated with the instance.

Reviewer: Без артикля

> ## Import

Reviewer: Разел таймаутов не там
